eras/alonzo/impl/testlib/Test/Cardano/Ledger/Alonzo/Arbitrary.hs

```diff
@@ -0,0 +1,475 @@
+import Cardano.Ledger.Alonzo.Scripts (
+  AlonzoScript (..),
+  CostModel,
+  CostModels (..),
+  ExUnits (..),
+  Prices (..),
+  Tag (..),
+  mkCostModel,
+  mkCostModelsLenient,
+ )
+import Cardano.Ledger.Alonzo.Scripts.Data (
+  BinaryData,
+  Data (..),
+  Datum (..),
+  dataToBinaryData,
+  hashData,
+ )
+import Cardano.Ledger.Alonzo.Tx (AlonzoTx (AlonzoTx), IsValid (IsValid), ScriptIntegrity (ScriptIntegrity), ScriptPurpose (..), getLanguageView)
+import Cardano.Ledger.Alonzo.TxAuxData (
+  AlonzoTxAuxData (..),
+  mkAlonzoTxAuxData,
+ )
+import Cardano.Ledger.Alonzo.TxBody (AlonzoTxBody (AlonzoTxBody))
+import Cardano.Ledger.Alonzo.TxOut (AlonzoTxOut (AlonzoTxOut))
+import Cardano.Ledger.Alonzo.TxWits (AlonzoTxWits (AlonzoTxWits), RdmrPtr (RdmrPtr), Redeemers (Redeemers), TxDats (TxDats))
+import Cardano.Ledger.BaseTypes (StrictMaybe)
+import Cardano.Ledger.Binary (DecCBOR, EncCBOR)
+import Cardano.Ledger.Crypto (Crypto)
+import Cardano.Ledger.Language (Language (..), nonNativeLanguages)
+import Cardano.Ledger.Shelley.LedgerState (PPUPPredFailure)
+import Cardano.Ledger.Shelley.Rules (PredicateFailure, ShelleyUtxowPredFailure)
+import Cardano.Ledger.Shelley.TxAuxData (Metadatum)
+import Control.Monad (replicateM)
+import Data.ByteString.Short (ShortByteString)
+import Data.Either (fromRight)
+import Data.Functor.Identity (Identity)
+import Data.Int (Int64)
+import Data.List.NonEmpty (NonEmpty ((:|)))
+import Data.Map (Map)
+import qualified Data.Map.Strict as Map
+import qualified Data.Set as Set
+import Data.Text (pack)
+import Data.Word (Word8)
+import Numeric.Natural (Natural)
+import qualified PlutusLedgerApi.Test.Examples as Plutus (
+  alwaysFailingNAryFunction,
+  alwaysSucceedingNAryFunction,
+  saltFunction,
+ )
+import qualified PlutusLedgerApi.V1 as PV1
+import qualified PlutusLedgerApi.V2 as PV2
+import PlutusPrelude (enumerate)
+import Test.Cardano.Ledger.Common
+import Test.Cardano.Ledger.Mary.Arbitrary (genMintValues)
+
+instance Era era => Arbitrary (Data era) where
+  arbitrary = Data <$> arbitrary
+
+instance Era era => Arbitrary (BinaryData era) where
+  arbitrary = dataToBinaryData <$> arbitrary
+
+instance Arbitrary PV1.Data where
+  arbitrary = resize 5 (sized gendata)
+    where
+      gendata n
+        | n > 0 =
+            oneof
+              [ PV1.I <$> arbitrary
+              , PV1.B <$> arbitrary
+              , PV1.Map <$> listOf ((,) <$> gendata (n `div` 2) <*> gendata (n `div` 2))
+              , PV1.Constr
+                  <$> fmap fromIntegral (arbitrary :: Gen Natural)
+                  <*> listOf (gendata (n `div` 2))
+              , PV1.List <$> listOf (gendata (n `div` 2))
+              ]
+      gendata _ = oneof [PV1.I <$> arbitrary, PV1.B <$> arbitrary]
+
+instance
+  ( Script era ~ AlonzoScript era
+  , Arbitrary (Script era)
+  , Arbitrary Metadatum
```

Comment:
This shouldn't be necessary. It might be that an instance is not in scope, which means it should be imported.
```suggestion
```